crates/core/tests/client.rs

```diff
@@ -1,44 +1,45 @@
+mod fixtures;
 mod tests {
 
     use ctor::ctor;
-    use ethers::abi::Token;
-    use ethers::types::Address as EthersAddress;
     use kakarot_rpc_core::client::api::KakarotEthApi;
     use kakarot_rpc_core::models::balance::{TokenBalance, TokenBalances};
     use kakarot_rpc_core::models::felt::Felt252Wrapper;
-    use kakarot_rpc_core::test_utils::deploy_helpers::{
-        create_raw_ethereum_tx, ContractDeploymentArgs, KakarotTestEnvironment,
-    };
+    use kakarot_rpc_core::test_utils::deploy_helpers::create_raw_ethereum_tx;
     use reth_primitives::{Address, BlockId, BlockNumberOrTag, U256};
+    use rstest::*;
     use starknet::core::types::FieldElement;
     use tracing_subscriber::FmtSubscriber;
 
+    use crate::fixtures::{kakarot_test_env_ctx, KakarotTestEnvironmentContext, TestContext};
+
     #[ctor]
     fn setup() {
         let subscriber = FmtSubscriber::builder().with_max_level(tracing::Level::ERROR).finish();
         tracing::subscriber::set_global_default(subscriber).expect("setting tracing default failed");
     }
 
-    #[tokio::test]
-    async fn test_rpc_should_not_raise_when_eoa_not_deployed() {
+    #[rstest]
+    #[tokio::test(flavor = "multi_thread")]
```

Comment:
is it mandatory to add `#[tokio::test(flavor = "multi_thread")]`? I thought that was the default setting for tokio::test

Reply:
Was having issues initially with just `#[tokio::test]` and I found this comment on an issue https://github.com/la10736/rstest/issues/141#issuecomment-1362888743 saying it can be fixed by explicitly setting `multi_thread`